File: TiaCodegen/Commands/Comparisons/InRangeCall.cs
using System.Linq;
using TiaCodegen.Commands.Functions.Base;
using TiaCodegen.Enums;
using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands.Comparisons
{
    public class InRangeCall : SystemFunctionCall 
    {
        public InRangeCall(
            IOperationOrSignal min,
            IOperationOrSignal @in,
            IOperationOrSignal max,
            IOperationOrSignal @out,
            IOperationOrSignal eno = null) : base("InRange", eno)
        {
            DisableEno = false;
            Interface["min"] = new IOperationOrSignalDirectionWrapper(min, Direction.Input);
            Interface["in"] = new IOperationOrSignalDirectionWrapper(@in, Direction.Input);
            Interface["max"] = new IOperationOrSignalDirectionWrapper(max, Direction.Output);
            Interface["out"] = new IOperationOrSignalDirectionWrapper(@out, Direction.Output);

            Children.AddRange(Interface.Values.Where(x => x.OperationOrSignal != null).Select(x => x.OperationOrSignal));
        }
    }
}

File: TiaCodegen-ts/src/Commands/Comparisons/InRangeCall.ts
import { IOperationOrSignal } from '../../Interfaces/IOperationOrSignal';
import { IOperationOrSignalDirectionWrapper } from '../../Interfaces/IOperationOrSignalDirectionWrapper';
import { Direction } from '../../Enums/Direction';
import { SystemFunctionCall } from '../Functions/Base/SystemFunctionCall';

export class InRangeCall extends SystemFunctionCall {
    constructor(
        min: IOperationOrSignal,
        inParam: IOperationOrSignal,
        max: IOperationOrSignal,
        out: IOperationOrSignal,
        eno: IOperationOrSignal | null = null,
    ) {
        super('InRange', eno);
        this.disableEno = false;
        this.iface['min'] = new IOperationOrSignalDirectionWrapper(min, Direction.Input);
        this.iface['in'] = new IOperationOrSignalDirectionWrapper(inParam, Direction.Input);
        this.iface['max'] = new IOperationOrSignalDirectionWrapper(max, Direction.Output);
        this.iface['out'] = new IOperationOrSignalDirectionWrapper(out, Direction.Output);

        for (const w of Object.values(this.iface)) {
            if (w.operationOrSignal !== null) this.children.push(w.operationOrSignal);
        }
    }
}
